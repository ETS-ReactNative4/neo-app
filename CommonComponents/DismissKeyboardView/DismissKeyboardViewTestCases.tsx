import React from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import { TextInput, StyleSheet } from "react-native";
import DismissKeyboardView from "./DismissKeyboardView";

const styles = StyleSheet.create({
  viewContainer: {
    flex: 1
  },
  inputField: {
    color: "black"
  }
});

const DismissKeyboardViewTestCases: ITestCase[] = [
  {
    title: "Dismiss Keyboard View",
    Component: (
      <DismissKeyboardView style={styles.viewContainer}>
        <TextInput
          style={styles.inputField}
          placeholder={"Tap anywhere on screen to hide keyboard..."}
          placeholderTextColor={"black"}
        />
      </DismissKeyboardView>
    )
  }
];

export default DismissKeyboardViewTestCases;
