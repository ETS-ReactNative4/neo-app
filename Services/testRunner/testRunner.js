import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
