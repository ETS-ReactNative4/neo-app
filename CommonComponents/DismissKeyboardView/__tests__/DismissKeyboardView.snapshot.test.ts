import DismissKeyboardViewTestCases from "../DismissKeyboardViewTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

DismissKeyboardViewTestCases.map(testRunner);
