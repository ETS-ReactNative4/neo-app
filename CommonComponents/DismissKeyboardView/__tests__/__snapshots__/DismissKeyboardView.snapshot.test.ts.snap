// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dismiss Keyboard View 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Tap anywhere on screen to hide keyboard..."
      placeholderTextColor="black"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "black",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</TouchableWithoutFeedback>
`;
